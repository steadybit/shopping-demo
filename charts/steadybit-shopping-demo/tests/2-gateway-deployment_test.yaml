templates:
  - 2-gateway-deployment.yml
chart:
  appVersion: v0.0.0
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot replica count
    set:
      gateway:
        replicaCount: 2
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with service type clusterIp
    set:
      gateway:
        service:
          type: ClusterIP
    asserts:
      - matchSnapshot: { }

