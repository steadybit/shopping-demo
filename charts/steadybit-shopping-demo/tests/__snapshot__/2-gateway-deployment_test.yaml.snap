manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-gateway
        run: gateway
        service-tier: "1"
        tags.datadoghq.com/service: shop-gateway
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: gateway
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: gateway-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-gateway
            run: gateway-exposed
            service-tier: "1"
            tags.datadoghq.com/service: shop-gateway
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - gateway-exposed
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: JAVA_TOOL_OPTIONS
                  value: -Dsun.net.inetaddr.ttl=0
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.fashion
                  value: http://fashion-bestseller:8082/products
                - name: rest.endpoint.toys
                  value: http://toys-bestseller:8081/products
                - name: rest.endpoint.hotdeals
                  value: http://hot-deals:8083/products
                - name: rest.endpoint.checkout
                  value: http://checkout:8085/checkout
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/gateway:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: gateway
              ports:
                - containerPort: 8080
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /work
                  name: work-dir
          serviceAccountName: steadybit-demo
          volumes:
            - emptyDir:
                sizeLimit: 256Mi
              name: work-dir
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: gateway
      name: gateway
      namespace: NAMESPACE
    spec:
      ports:
        - port: 80
          protocol: TCP
          targetPort: 8080
      selector:
        run: gateway-exposed
      type: LoadBalancer
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: gateway-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: gateway-exposed
manifest should match snapshot replica count:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-gateway
        run: gateway
        service-tier: "1"
        tags.datadoghq.com/service: shop-gateway
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: gateway
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: gateway-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-gateway
            run: gateway-exposed
            service-tier: "1"
            tags.datadoghq.com/service: shop-gateway
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - gateway-exposed
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: JAVA_TOOL_OPTIONS
                  value: -Dsun.net.inetaddr.ttl=0
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.fashion
                  value: http://fashion-bestseller:8082/products
                - name: rest.endpoint.toys
                  value: http://toys-bestseller:8081/products
                - name: rest.endpoint.hotdeals
                  value: http://hot-deals:8083/products
                - name: rest.endpoint.checkout
                  value: http://checkout:8085/checkout
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/gateway:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: gateway
              ports:
                - containerPort: 8080
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /work
                  name: work-dir
          serviceAccountName: steadybit-demo
          volumes:
            - emptyDir:
                sizeLimit: 256Mi
              name: work-dir
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: gateway
      name: gateway
      namespace: NAMESPACE
    spec:
      ports:
        - port: 80
          protocol: TCP
          targetPort: 8080
      selector:
        run: gateway-exposed
      type: LoadBalancer
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: gateway-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: gateway-exposed
manifest should match snapshot with service type clusterIp:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-gateway
        run: gateway
        service-tier: "1"
        tags.datadoghq.com/service: shop-gateway
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: gateway
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: gateway-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-gateway
            run: gateway-exposed
            service-tier: "1"
            tags.datadoghq.com/service: shop-gateway
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - gateway-exposed
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: JAVA_TOOL_OPTIONS
                  value: -Dsun.net.inetaddr.ttl=0
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.fashion
                  value: http://fashion-bestseller:8082/products
                - name: rest.endpoint.toys
                  value: http://toys-bestseller:8081/products
                - name: rest.endpoint.hotdeals
                  value: http://hot-deals:8083/products
                - name: rest.endpoint.checkout
                  value: http://checkout:8085/checkout
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/gateway:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: gateway
              ports:
                - containerPort: 8080
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8080
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /work
                  name: work-dir
          serviceAccountName: steadybit-demo
          volumes:
            - emptyDir:
                sizeLimit: 256Mi
              name: work-dir
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: gateway
      name: gateway
      namespace: NAMESPACE
    spec:
      ports:
        - port: 80
          protocol: TCP
          targetPort: 8080
      selector:
        run: gateway-exposed
      type: ClusterIP
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: gateway-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: gateway-exposed
